CAS-479: bedspace search layout

froddd opened the pull request:
# Context

https://dsdmoj.atlassian.net/browse/CAS-479

# Changes in this PR

The search form and results are now split into 2 templates in preparation of other search parameters being added to the form. The layout and styles of the search results have been updated to match latest designs.

Tests for the bedspace search controller have been rejigged to separate concerns between form page and results page better.

## Screenshots of UI changes

### Before

#### Search form

<img width="714" alt="Screenshot 2024-07-30 at 16 56 15" src="https://github.com/user-attachments/assets/c4b39945-20a8-4525-8bbd-7c5a5d9e79ee">

---

#### Results page

<img width="1199" alt="Screenshot 2024-07-30 at 16 55 59" src="https://github.com/user-attachments/assets/36d548ec-8ff3-44a5-8a14-17d62279e288">

---

### After

#### Search form

<img width="708" alt="Screenshot 2024-07-30 at 16 55 00" src="https://github.com/user-attachments/assets/9325773f-1355-4fb2-97e8-ab7b8faa56bb">

---

#### Results page

<img width="1192" alt="Screenshot 2024-08-05 at 12 03 38" src="https://github.com/user-attachments/assets/e2439d46-5990-46fd-97a6-11451e070882">


---

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

<!-- Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible. You can monitor deployment failures in CircleCI
itself and application errors are found in
[Sentry](https://ministryofjustice.sentry.io/issues/?project=4504129156218880&referrer=sidebar&statsPeriod=24h).
Both events should be automatically sent to our [Slack
channel](https://mojdt.slack.com/archives/C048BJS7S2F). -->

Commit:
Tidy up test suite

This separates the concerns of showing the form and showing results to
accommodate coming changes to layout.

Commit:
Split form and results into separate templates

Extract form fields into reusable partial.

Commit:
Update search form layout

Form is now on teh left of results, and includes correct fieldsets
around date fields. Buttons are updated as per design, with form page
showing a 'Clear all' button and results page showing a 'Clear filters'
link.

aliuk2012 commented on server/views/temporary-accommodation/bedspace-search/results.njk:
```diff
@@ -0,0 +1,128 @@
+
+{% extends "../../partials/layout.njk" %}
+
+{% set pageTitle = "Search for available bedspaces - " + applicationName %}
```
🤔  Would it be worth mentioning how many results are returned in the page title? Ideally the page title should update if the page content could potentially be different ie. search page and results page

Commit:
Update bedspace search results layout and styling

- Use Filter component for left-hand form
- Use card for each result
- Use heading for no results, with tweaked wording

aliuk2012 commented on assets/sass/components/_filter.scss:
```diff
@@ -0,0 +1,9 @@
+ * https://design-patterns.service.justice.gov.uk/components/filter/
+ */
+
+.moj-filter__header {
```
```suggestion
.moj-filter__header--light-grey {
```

Could we use BEM modifer class names instead of modifying the component class directly? Its easier to identify that this is our own implementation although if we are modifying components we should really try to push the changes back to moj frontend else moj will end up have 1000 slightly different component instead of a single reusable and consistent component.

froddd replied:
That would be ideal, but unfortunately the macro doesn't enable the use of any modifier classes, hence the extra rules added to the source class (https://github.com/ministryofjustice/moj-frontend/blob/main/src/moj/components/filter/template.njk#L3). I'm open to any other suggestion, I don't like it much myself this way!

aliuk2012 replied:
😬 not much you can customise with that component. 

Could we make a contribution? we could add custom classes to the component itself or to the header something like how govuk do it https://github.com/alphagov/govuk-frontend/blob/main/packages/govuk-frontend/src/govuk/components/back-link/template.njk#L3

its a pitty the head/body parts of the component where not split out as a micro components themselves where you could pass in custom css class names or attributes to the different part.

froddd replied:
https://github.com/ministryofjustice/moj-frontend/issues/611

aliuk2012 commented on assets/sass/components/_result-card.scss:
```diff
@@ -0,0 +1,18 @@
+
+    & .govuk-heading-m {
+      margin-bottom: 0;
+    }
```
Is this needed?

Could we not add `govuk-!-margin--bottom-0`  to the headings that need margin 0? Theres a really good write up https://design-system.service.gov.uk/get-started/extending-and-modifying-components/

aliuk2012 commented on assets/sass/components/_result-card.scss:
```diff
@@ -0,0 +1,18 @@
+  }
+
+  &__body {
+    padding: govuk-spacing(4) govuk-spacing(4) 0;
```
```suggestion
    @include govuk-responsive-padding(4, "top");
    @include govuk-responsive-padding(4, "left");
    @include govuk-responsive-padding(4, "right");
    @include govuk-responsive-padding(0, "bottom");
```

Did you want to try this? More details https://frontend.design-system.service.gov.uk/sass-api-reference/#govuk-responsive-padding and theres a matching margin mixin too. It might make a padding a bit smoother when viewport changes. 

https://design-system.service.gov.uk/styles/spacing/#responsive-spacing

aliuk2012 commented on assets/sass/components/_result-card.scss:
```diff
@@ -0,0 +1,18 @@
+.result-card {
+  border: 1px solid govuk-colour('mid-grey');
+  margin-bottom: govuk-spacing(6);
```
```suggestion
  @include govuk-responsive-margin(6, "bottom");
```

aliuk2012 commented on assets/sass/components/_result-card.scss:
```diff
@@ -0,0 +1,18 @@
+  &__header {
+    border-bottom: 1px solid govuk-colour('mid-grey');
+    background-color: govuk-colour("light-grey");
+    padding: govuk-spacing(4);
```
```suggestion
    @include govuk-responsive-padding(4, "all");
    border-bottom: 1px solid govuk-colour('mid-grey');
    background-color: govuk-colour("light-grey");
```

aliuk2012 commented on server/views/temporary-accommodation/bedspace-search/results.njk:
```diff
@@ -0,0 +1,143 @@
+
+        <div class="govuk-grid-column-two-thirds">
+            {% if results.results.length === 0 %}
+                <p class="govuk-body-l">
```
🤔  should this be h2? because its the heading for "No results" part of the page.

aliuk2012 commented on server/views/temporary-accommodation/bedspace-search/results.njk:
```diff
@@ -0,0 +1,143 @@
+                </p>
+                <p class="govuk-body-l">Try changing one or more of your search criteria to get more results.</p>
+            {% else %}
+                {% for result in results.results %}
```
🤔 I'm wondering if we can improve this as its tricky dealing with `results.results`. Its tricky to work out what exactly is results of results by looking only at the view

Commit:
Ensure place context remains when clearing filters
